### ISSUE TITLE
array-bracket-spacing fix removes comments following space

### ISSUE THREAD

**Issue by grantrules** [eslint/eslint#13701]
**Tell us about your environment**

* **ESLint Version: 7.9.0**
* **Node Version: 14.10.0**
* **npm Version: 6.14.8**

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

Default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  env: {
    es2021: true,
    node: true,
  },
  extends: [
    'airbnb-base',
  ],
  parserOptions: {
    ecmaVersion: 12,
    sourceType: 'module',
  },
  rules: {
  },
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

index.js:

```js
const [ /* one */, two] = ['hello', 'world'];
```

```bash
npx eslint --fix index.js
```

**What did you expect to happen?**

I expected the result would be:

```js
const [/* one */, two] = ['hello', 'world'];
```

**What actually happened? Please include the actual, raw output from ESLint.**

Instead the result was:

```js
const [, two] = ['hello', 'world'];
```

The output from ESLint was simply:

```1:10  error  'two' is assigned a value but never used  no-unused-vars```

**Are you willing to submit a pull request to fix this bug?**

No

**Comment by anikethsaha** [comment]
Thanks for the report. 

This is a known bug. We are working on it. 

Ref: #13319
